### PR TITLE
Fix swap_dims() index names (issue #3748)

### LINKED ISSUE #3748: `swap_dims()` incorrectly changes underlying index name
#### MCVE Code Sample

```python
import xarray as xr

# create data array with named dimension and named coordinate
x = xr.DataArray([1], {"idx": [2], "y": ("idx", [3])}, ["idx"], name="x")

# what's our current index? (idx, this is fine)
x.indexes
# prints "idx: Int64Index([2], dtype='int64', name='idx')"

# swap dim so that y is our dimension, what's index now?
x.swap_dims({"idx": "y"}).indexes
# prints "y: Int64Index([3], dtype='int64', name='idx')"
```
The dimension name is appropriately swapped but the pandas index name is incorrect.

#### Expected Output
```python
# swap dim so that y is our dimension, what's index now?
x.swap_dims({"idx": "y"}).indexes
# prints "y: Int64Index([3], dtype='int64', name='y')"
```

#### Problem Description

This is a problem because running `x.swap_dims({"idx": "y"}).to_dataframe()` gives
a dataframe with columns `["x", "idx"]` and index `"idx"`. This gives ambiguous names and drops the original name, while the DataArray string representation gives no indication that this might be happening.

#### Output of ``xr.show_versions()``
<details>
# Paste the output here xr.show_versions() here
INSTALLED VERSIONS
------------------
commit: None
python: 3.8.1 | packaged by conda-forge | (default, Jan 29 2020, 15:06:10) 
[Clang 9.0.1 ]
python-bits: 64
OS: Darwin
OS-release: 18.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.5
libnetcdf: 4.7.3

xarray: 0.15.0
pandas: 0.25.3
numpy: 1.18.1
scipy: 1.4.1
netCDF4: 1.5.3
pydap: None
h5netcdf: None
h5py: 2.10.0
Nio: None
zarr: None
cftime: 1.0.4.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2.10.1
distributed: 2.10.0
matplotlib: None
cartopy: None
seaborn: None
numbagg: None
setuptools: 45.1.0.post20200119
pip: 20.0.2
conda: None
pytest: 5.3.5
IPython: 7.12.0
sphinx: None

</details>

### PR DESCRIPTION
**Pull request by jaicher**
- [x] Closes #3748
 - [x] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

**Comment by jaicher** [comment]
I added a simple test, but not sure it is correctly set up. I wasn't sure about how pytest fixtures worked, so I didn't know if I needed/how to set up a new fixture to test the bugfix.

**Comment by jaicher** [comment]
The 4 pytest failures per check are taking place in `tests/test_plot.py`. I suspect that these failures are unrelated to the pull request.

File: xarray/tests/test_dataarray.py [hunk 1]
```diff
@@ -4778,3 +4778,14 @@ def test_weakref():
     a = DataArray(1)
     r = ref(a)
     assert r() is a
+
+
+def test_swap_dims_index_name():
```
Comment (dcherian) [inline xarray/tests/test_dataarray.py]: Can we modify `test_swap_dims` in this file instead? We should also modify `test_swap_dims` in  `test_dataset.py` too.

Can you also add a MultiIndex test case to `test_swap_dims` in both files.

Comment (jaicher) [reply xarray/tests/test_dataarray.py]: Done -- please let me know if there any other changes you think we need. Thanks for the quick review!

**Comment by jaicher** [comment]
We have different test failures than last time. They appear to be happening in tests with time indexes this time. I suspect these failures are again unrelated to the pull request.

**Comment by dcherian** [comment]
test failure is unrelated. Thanks @jaicher

**Comment by max-sixty** [comment]
Thanks a lot @jaicher 

Thanks for all the recent reviews @dcherian ; I'll attempt to contribute a bit more...

**Pull request merged**